react/components/ProductBrand/index.js

```diff
@@ -0,0 +1,177 @@
+
+            return (
+              <BrandContainer>
+                <div
+                  style={{
+                    // width,
+                    height,
+                  }}
+                >
+                  {/** TODO: Use a smarter Image component that handles VTEX image resizing etc. */}
+                  <img
+                    className={generateBlockClass(
+                      styles.productBrandImage,
+                      blockClass
+                    )}
+                    src={changeImageUrlSize(
+                      `/arquivos/ids${imageUrl}`,
+                      // width ? width * dpi : undefined,
+                      undefined,
+                      height ? height * dpi : undefined
+                    )}
+                    style={{
+                      // width: width || 'auto',
+                      height: height || 'auto',
+                    }}
+                  />
+                </div>
+              </BrandContainer>
+            )
+          }
+
+          /** If imageUrl is exactly null, it means no brand
+           * logo was uploaded. So it falls back to the brand name,
+           * if said fallback is enabled. */
+          if (imageUrl === null && fallbackToText) {
+            return <BrandContainer>{brandNameElement}</BrandContainer>
+          }
+        }
+
+        /** Renders an empty spacer in the size of the preferred
+         * type (i.e. if most brands have no logo, it should render
+         * a placeholder in the size of the brand name text; if
+         * most brands do have a logo, it should render a spacer
+         * in the size of the predefined image size)
+         */
+        if (loadingPlaceholder === DISPLAY_MODE.LOGO) {
+          return (
+            <BrandContainer>
+              <div
+                style={{
+                  // width,
+                  height,
+                }}
+              />
+            </BrandContainer>
+          )
+        }
+        if (loadingPlaceholder === DISPLAY_MODE.TEXT && fallbackToText) {
+          return (
+            <BrandContainer>
+              <span className="o-0">{brandNameElement}</span>
+            </BrandContainer>
+          )
+        }
+
+        return null
+      }}
+    </Query>
+  )
+}
+
+ProductBrand.propTypes = {
+  /** Brand name */
+  brandName: PropTypes.string,
+  /** Whether it should be displayed as a logo or as a text */
+  displayMode: PropTypes.oneOf(Object.values(DISPLAY_MODE)),
+  /** Whether the loading placeholder should have the size of the logo or the text */
+  loadingPlaceholder: PropTypes.oneOf(Object.values(DISPLAY_MODE)),
+  /** Whether it should display the name of the brand if there is no logo */
+  fallbackToText: PropTypes.bool,
+  /** List of brands that should be hidden, if any */
+  excludeBrands: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.number, PropTypes.string])
+  ),
+  /** Height of the logo */
+  height: PropTypes.nuber,
```

Comment:
![image](https://user-images.githubusercontent.com/284515/59948198-c2996d80-9445-11e9-9e46-e822e63ef586.png)
```suggestion
  height: PropTypes.number,
```

Reply:
😅